networks/archive/1a_transformation_correlation_analysis.py

```diff
@@ -0,0 +1,135 @@
+import umap
+import random
+import numpy as np
+from sklearn import preprocessing
+from core_acc_modules import paths
+
+# Load expression data
+pao1_compendium_filename = paths.PAO1_COMPENDIUM
+pa14_compendium_filename = paths.PA14_COMPENDIUM
+
+pao1_compendium = pd.read_csv(pao1_compendium_filename, sep="\t", header=0, index_col=0)
+pa14_compendium = pd.read_csv(pa14_compendium_filename, sep="\t", header=0, index_col=0)
+
+print(pao1_compendium.shape)
+pao1_compendium.head()
+
+print(pa14_compendium.shape)
+pa14_compendium.head()
+
+# ## Correlation of raw gene expression data
+#
+# Use this as a comparison to see how the correlations are changed after each correction method
+
+# Correlation
+pao1_corr_original = pao1_compendium.corr()
+pa14_corr_original = pa14_compendium.corr()
+
+# Plot heatmap
+plt.figure(figsize=(20, 20))
+o1 = sns.clustermap(pao1_corr_original.abs(), cmap="viridis")
+o1.fig.suptitle("Correlation of raw PAO1 genes", y=1.05)
+
+# Plot heatmap
+plt.figure(figsize=(20, 20))
+o2 = sns.clustermap(pa14_corr_original.abs(), cmap="viridis")
+o2.fig.suptitle("Correlation of raw PA14 genes", y=1.05)
+
+# ## Log transform + correlation
+
+# log transform data
+# Note: add 1 to avoid -inf and so 0 corresponds to those with 0 counts
+pao1_compendium_log10 = np.log10(1 + pao1_compendium)
+pa14_compendium_log10 = np.log10(1 + pa14_compendium)
+
+sns.displot(pao1_compendium_log10["PA0001"])
+
+# Correlation
+pao1_corr_log10 = pao1_compendium_log10.corr()
+pa14_corr_log10 = pa14_compendium_log10.corr()
+
+# Plot heatmap
+plt.figure(figsize=(20, 20))
+h1 = sns.clustermap(pao1_corr_log10.abs(), cmap="viridis")
+h1.fig.suptitle("Correlation of log10 transformed PAO1 genes", y=1.05)
+
+plt.figure(figsize=(20, 20))
+h2 = sns.clustermap(pa14_corr_log10.abs(), cmap="viridis")
+h2.fig.suptitle("Correlation of log10 transformed PA14 genes", y=1.05)
+
+# ## 0-1 normalize + correlation
+
+# +
+# 0-1 normalize per gene
+scaler = preprocessing.MinMaxScaler()
+
+pao1_scaled = scaler.fit_transform(pao1_compendium)
+pao1_scaled_df = pd.DataFrame(
+    pao1_scaled, columns=pao1_compendium.columns, index=pao1_compendium.index
+)
+
+scaler2 = preprocessing.MinMaxScaler()
+pa14_scaled = scaler2.fit_transform(pa14_compendium)
+pa14_scaled_df = pd.DataFrame(
+    pa14_scaled, columns=pa14_compendium.columns, index=pa14_compendium.index
+)
+# -
+
+sns.displot(pao1_scaled_df["PA0001"])
+
+# Correlation
+pao1_corr_normalized = pao1_scaled_df.corr()
+pa14_corr_normalized = pa14_scaled_df.corr()
+
+# Plot heatmap
+plt.figure(figsize=(20, 20))
+h3 = sns.clustermap(pao1_corr_normalized.abs(), cmap="viridis")
+h3.fig.suptitle("Correlation of 0-1 normalized PAO1 genes", y=1.05)
+
+plt.figure(figsize=(20, 20))
+h4 = sns.clustermap(pa14_corr_normalized.abs(), cmap="viridis")
+h4.fig.suptitle("Correlation of 0-1 normalized PA14 genes", y=1.05)
+
+# **Takeaway:**
+# Here we are using the Pearson correlation metric, which is defined by the formula:
```

Comment:
I can't remember - have you considered trying [Spearman rank correlation](https://en.wikipedia.org/wiki/Spearman%27s_rank_correlation_coefficient) instead of Pearson? That could help in the cases where you have varying scales/ranges (Spearman is essentially the Pearson correlation of the rank vectors).

I think people sometimes use Spearman to construct co-expression networks (see, e.g. [this paper](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3511551/)).

Not that you necessarily need more things to try - seems like the log-transform + SPELL approach is giving you what you want, so feel free to stick with that if you want to.

Reply:
I did try Spearman correlation earlier. See [slide 10](https://docs.google.com/presentation/d/10Pr_oD5jL8GxCHLg1Qb16A6br31E2COgZIThjnj5ZyI/edit#slide=id.gd398a77b8a_0_4).

It does look like Spearman can help a bit, but I don't think results should change too much if we switch.